exonum/src/proto/mod.rs

```diff
@@ -335,6 +337,77 @@ where
     }
 }
 
+impl<K, V> ProtobufConvert for exonum_merkledb::MapProof<K, V>
+where
+    K: BinaryKey,
+    V: BinaryValue,
+    Vec<(K, Option<V>)>: FromIterator<(<K as ToOwned>::Owned, Option<V>)>,
```

Comment:
This is a weird type constraint. Cannot it be simplified to `K: From<<K as ToOwned>::Owned>`, which seems to be logically equivalent to `K: ToOwned<Owned = K>`, or , logically speaking, `K: 'static` (although I'm not quite sure that the compiler will be able to correctly reason about the latter).